### DISCUSSION TITLE
How can I generate diifferent pore size distribution images? and save it to my computer?

### DISCUSSION THREAD

**Discussion by himanshu-sahu674** [PMEAL/porespy#734] [Q&A] [answered]
I want to generate a bunch of image something around 200 with different pore size distribution with some realistic porosity range. How can I generate the image of different pore size distribution and also want to save them in my local computer storage.

**Comment by jgostick** [comment, marked as answer]
I am very sorry for the delay in responding...I didn't notice this until just now.  You've probably already solved your problem and moved on, but here is an answer anyway...

The generators module contains many options that should work.  For instance, if you create all images with 200^3 size, then you can use `overlapping_sphere` with different porosity and radius, or you can use `blobs` with different porosity and blobiness.  The size of the pore will depend on what you assume as the voxel size.  These generators are sizeless, but you can assume that 200 voxels is 800 um, thus each voxel is 4um, then you'd put that into other functions as the `voxel_size`.

**Reply by jgostick** [reply]
I should also provide a link to the use of `blobs`: https://porespy.org/examples/generators/tutorials/making_blobs.html

**Comment by himanshu-sahu674** [comment]
Thank You Sir for your response.

Actually I'm a Petroleum Engineer persuing master's from Indian Institute
of Technology (Indian School of Mines).
And for my project work I have to train the CNN model to predict the
permeability from CT images and for that purpose I need to generate some
database with some generic image (generated by Porespy) then calculate
permeability of that image using OpenPNM and after generation this
synthetic data. We have some original field data of Volvo Field and in that
data we have CT scan images in a pdf format ( I already took the screenshot
of that image and the size of that images are 600x400). And also I have
generated the image by using blob generator with shape = [600x400].
So basically we have 2 different type of images and corresponding porosity
and permeability.
1). Generated image from porespy by using blobs
2). Volve field image with porosity and permeability

For training and testing purpose we want to use mix data some synthetic
image and some real images.

But sir I'm stucked at the point how to scale all the images in same unit,
I don't know much about "voxel size" I simplely have taken 5.34um as had
taken in Berea Sandstone case for predicting permeability. And if I'm
changing the voxel size my permeability is also changing.

How can I understand this thing? Please suggest something what should I do
now?

Thank you
Himanshu

**Reply by jgostick** [reply]
Permeability should change with voxel size.  If voxel size is low, then pores/throats are small, and flow is restricted.  Conversely if large voxels, then the everything is 'bigger' and flow is easeir.  This relationship is described by the [Hagan-Poiseuille](https://en.wikipedia.org/wiki/Hagen%E2%80%93Poiseuille_equation) and also captured by the Carman-Kozeny correlation which has particle size in it.  Basically, you can use the same image, with different voxel size, and get extra training data for your AI since the images will have different pores sizes.

**Comment by himanshu-sahu674** [comment]
Hello Sir,
I have microphotograph of porous rock at a particular depth and also calculated permeability & porosity by core analysis at that depth but I don't know the scale (um/pixel) of that image.

Now while I'm generating synthetic images with the help of porespy 's blob generator with porosity value in between 10-25%, and extracting pore network using snow2 there is a parameter called voxel_size which is by default showing 1.

So sir my questions are -

1. How can we know the scale (um/pixel) of original image if we don't know the CT scan resolution?
2. In snow algorithm by default voxel_size = 1, so can we think it is 1um/pixel of generated image?

Because I want both the images original and generated in same scale for train the model that will predict the permeability.

PFA original image with porosity = 10.8 % and permeability k_horizontal = 21.4 measured by laboratory core analysis.


![(3837)](https://user-images.githubusercontent.com/85443615/202106846-05958ffb-867e-4aa8-89e5-31fa1a38971d.png)

Can you please suggest me what should I do to at this point of my project where I got stuck to build the database.

Thank You
